core/src/main/java/com/linecorp/armeria/internal/common/websocket/WebSocketUtil.java

```diff
@@ -46,8 +51,66 @@ public static boolean isHttp1WebSocketUpgradeRequest(RequestHeaders headers) {
         // Connection: Upgrade
         // ...
         return headers.method() == HttpMethod.GET &&
-               HttpHeaderValues.UPGRADE.contentEqualsIgnoreCase(headers.get(HttpHeaderNames.CONNECTION)) &&
-               HttpHeaderValues.WEBSOCKET.contentEqualsIgnoreCase(headers.get(HttpHeaderNames.UPGRADE));
+               containsValue(headers, HttpHeaderNames.CONNECTION, HttpHeaderValues.UPGRADE, true) &&
+               containsValue(headers, HttpHeaderNames.UPGRADE, HttpHeaderValues.WEBSOCKET, true);
+    }
+
+    // Forked from https://github.com/netty/netty/blob/4.1/codec-http/src/main/java/io/netty/handler/codec/http/HttpHeaders.java#L1597-L1651
+    private static boolean containsValue(RequestHeaders headers, CharSequence name,
```

Comment:
Q: Is it worth adding this to `HttpHeaders` as a public API?

Reply:
That's a good suggestion. 👍 Let me handle it in a separate PR because `QueryParamGetters` needs a different logic for the method. 😉